internal/cmd/run.go

```diff
@@ -41,37 +43,45 @@ func runCmd() *cobra.Command {
 				return err
 			}
 
-			if err := replace(replaceScripts, block.Lines()); err != nil {
-				return err
-			}
+			return runBlockCmd(block, cmd, &opts)
+		},
+	}
 
-			if id, ok := shellID(); ok && runner.IsShell(block) {
-				return executeInShell(id, block)
-			}
+	setDefaultFlags(&cmd)
 
-			executable, err := newExecutable(cmd, block)
-			if err != nil {
-				return err
-			}
+	cmd.Flags().BoolVar(&opts.dryRun, "dry-run", false, "Print the final command without executing.")
+	cmd.Flags().StringArrayVarP(&opts.replaceScripts, "replace", "r", nil, "Replace instructions using sed.")
 
-			ctx, cancel := ctxWithSigCancel(cmd.Context())
-			defer cancel()
+	return &cmd
+}
 
-			if dryRun {
-				executable.DryRun(ctx, cmd.ErrOrStderr())
-				return nil
-			}
+func runBlockCmd(block *document.CodeBlock, cmd *cobra.Command, opts *runCmdOpts) error {
```

Comment:
To align the interface with `newExecutable()` you can change it to:

```suggestion
func runBlock(cmd *cobra.Command, block *document.CodeBlock, opts *runCmdOpts) error {
```